unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarsLODController.cs

```diff
@@ -0,0 +1,147 @@
+                            }
+                            else
+                            {
+                                enabled = false;
+                            }
+                        });
+        }
+
+        public void LateUpdate()
+        {
+            int listCount = avatarsList.Count;
+
+            bool applyVerticalMovement = Time.timeSinceLevelLoad - lastLODsVerticalMovementTime > LODS_VERTICAL_MOVEMENT_DELAY;
+            GameObject lodGO;
+            for (int i = 0; i < listCount; i++)
+            {
+                lodGO = avatarsList[i].avatarRenderer.lodRenderer.gameObject;
+                if (!lodGO.activeSelf)
+                    continue;
+
+                if (applyVerticalMovement)
+                    lodGO.transform.localPosition = new Vector3(lodGO.transform.localPosition.x, (lodGO.transform.localPosition.y > LODS_LOCAL_Y_POS ? LODS_LOCAL_Y_POS : LODS_LOCAL_Y_POS + LODS_VERTICAL_MOVEMENT), lodGO.transform.localPosition.z);
+
+                Vector3 previousForward = lodGO.transform.forward;
+                Vector3 lookAtDir = (lodGO.transform.position - CommonScriptableObjects.cameraPosition).normalized;
+
+                lookAtDir.y = previousForward.y;
+
+                lodGO.transform.forward = lookAtDir;
+            }
+
+            if (applyVerticalMovement)
+                lastLODsVerticalMovementTime = Time.timeSinceLevelLoad;
+        }
+
+        void OnMainPlayerReposition(Vector3 newPos, Vector3 previousPos) { UpdateAllLODs(); }
+
+        public void UpdateAllLODs()
+        {
+            if (!enabled)
+                return;
+
+            SortedList<float, AvatarShape> closeDistanceAvatars = new SortedList<float, AvatarShape>();
+            foreach (AvatarShape avatar in avatarsList)
+            {
+                float distanceToPlayer = Vector3.Distance(CommonScriptableObjects.playerUnityPosition.Get(), avatar.transform.position);
+                bool isInLODDistance = distanceToPlayer >= lodDistance;
+
+                if (isInLODDistance)
+                    ToggleLOD(avatar.avatarRenderer, true);
+                else
+                    closeDistanceAvatars.Add(distanceToPlayer, avatar);
+            }
+
+            int closeDistanceAvatarsCount = closeDistanceAvatars.Count;
+            AvatarShape currentAvatar;
+            for (var i = 0; i < closeDistanceAvatarsCount; i++)
+            {
+                currentAvatar = closeDistanceAvatars.Values[i];
+                bool isLOD = i >= maxNonLODAvatars;
+
+                ToggleLOD(currentAvatar.avatarRenderer, isLOD);
+            }
+        }
+
+        private void ToggleLOD(AvatarRenderer avatarRenderer, bool enabled)
+        {
+            avatarRenderer.lodRenderer.gameObject.SetActive(enabled);
+            avatarRenderer.SetVisibility(!enabled); // TODO: Resolve coping with AvatarModifierArea regarding this toggling
+        }
```

Comment:
This is just my obsessive disorder. Can we move private methods to the bottom of the class? pretty please? :)

Reply:
I reordered the methods, but left the Awake and LateUpdate on top because they are Unity hooks. Althought if I end up removing the monobehaviour functionality that will change.